daffodil-core/src/main/scala/org/apache/daffodil/grammar/HasStatementsGrammarMixin.scala

```diff
@@ -18,10 +18,11 @@
 package org.apache.daffodil.grammar
 
 import org.apache.daffodil.dsom.Term
+import org.apache.daffodil.dsom.DFDLNewVariableInstance
 
 trait HasStatementsGrammarMixin extends GrammarMixin { self: Term =>
 
-  private lazy val statementGrams = statements.map { _.gram(self) }
+  private lazy val statementGrams = statements.filter{ st => !st.isInstanceOf[DFDLNewVariableInstance] }.map { _.gram(self) }
```

Comment:
I think of a nve as a statement. So we should either rename statementGrams, or figure out whether NVE should now be included. I think it was filtered out as part of not being implemented before. We still had these combinators (grammar) being generated before, just no parsers/unparsers beyond that.

Reply:
So, the other possible statements are assert, discriminator, and setVariable. The reason NVI is filtered out is because they need to be processed before the other statements, or at least before setVariable statements. The test that Mike suggested elsewhere in the review actually triggered a bug as I was processing NVI after other statements.

Reply:
So then I suggest renaming these to the beforehandStatementGrams, and/or afterwardsStatementGrams to make the timing distinction clearer. A comment citing that part of the spec is always good.

Reply:
Actually, according to the spec we probably aren't doing the evaluation order of statement annotations correctly. Before NVI are supposed to be processed we are supposed to evaluate discriminators/asserts with test="pattern", but I think that may require significant changes and is outside the scope of this pull request.
I'll make the suggested changes of adding more clear names and comments.

Reply:
I think the parser order is right. In ElementCombinator1.scala, the parse() method first does pattern discrim/asserts. Then it does parseBegin which just creates the infoset element. Then it does eBeforeParser which is the NVE (and framing), and so on. So I think we're doing pattern stuff first.

Reply:
Ok, good to know.  I was just going off of the order the grammar seemed to be getting laid out in.

Reply:
Yeah, at one time I thought the grammar would implicitly orchestrate the order of all these things, and that was the original intent as you've recognized. 

But over time we've moved toward a style of just passing a bunch of Gram objects to a combinator which orchestrates them internally, differentiates parse from unparse behavior accordingly, etc.  This has the downside of every combinator having to deal with optimizing away in case of empty Gram objects for the various args, but the advantage of the resulting stack of combinators being better motivated than a bunch of sequencing operations only.